R/eesapi_url.R

```diff
@@ -0,0 +1,91 @@
+#' eesapi_url
+#'
+#' @param api_version EES API version
+#' @param endpoint Name of endpoint, can be get-summary, get-meta, get-data, query-data
+#' @param dataset_id ID of dataset to be connected to
+#' @param dataset_version Verson of dataset to be connected to
+#'
+#' @return The EES URL for connecting to the API
+#' @export
+#'
+#' @examples
+#' eesapi_url()
+eesapi_url <- function(
+    api_version = "1.0",
+    endpoint = NULL,
+    dataset_id = NULL,
+    dataset_version = NULL) {
+  # Check that the API version is valid
+  is_valid_api_version <- function(vapi) {
+    !grepl(
+      "[a-z_%+-]",
+      as.character(vapi),
+      ignore.case = TRUE
+    )
+  }
+
+  if (is_valid_api_version(api_version) == FALSE) {
+    stop(
+      "You have entered an invalid API version in the api_version argument.
+      This should be numerical values only."
+    )
+  }
+
+  # Check that the endpoint is either NULL or valid
+  is_valid_endpoint <- function(endpoint) {
+    endpoint %in% c("get-summary", "get-meta", "get-data", "query-data")
+  }
+
+  if (!is.null(endpoint)) {
+    if (is_valid_endpoint(endpoint) == FALSE) {
+      stop(
+        paste(
+          "You have entered an invalid endpoint, this should one of:",
+          "get-summary, get-meta, get-data or query-data"
+        )
+      )
+    }
+  }
+
+  is_valid_dataset_info <- function(dataset_id, dataset_version) {
+    !is.null(dataset_id) & (is.numeric(dataset_version) | is.null(dataset_version))
```

Comment:
Had a few issues with the | operator and long lines / line breaks styling weird with styler, so if that's a pain when shortening this, I would be tempted to rewrite to avoid stringing conditions with | or &

Reply:
Added a new line